packages/contracts/deploy/02_factory.ts

```diff
@@ -1,23 +1,35 @@
 import { HardhatRuntimeEnvironment } from "hardhat/types";
 import { DeployFunction, DeployOptions } from "hardhat-deploy/types";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { deployENS } from "@ethereum-waffle/ens";
+
+const deployEns = async (deployer: SignerWithAddress) => {
+    const ens = await deployENS(deployer);
+    await ens.createTopLevelDomain("test");
+    return ens.ens.address;
+};
 
 const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
-    const { deployments, getNamedAccounts } = hre;
+    const { deployments, ethers, getNamedAccounts } = hre;
     const { deployer } = await getNamedAccounts();
+    const [deployer_] = await ethers.getSigners();
```

Comment:
These names can be confusing. What is the difference between `deployer` and `deployer_`?

Reply:
Renamed the variable to `deployerSigner`